$MirOS: ports/www/firesomething/patches/patch-nsprpub_pr_src_misc_prnetdb_c,v 1.4 2008/12/05 16:26:11 bsiegert Exp $
$OpenBSD: patch-nsprpub_pr_src_misc_prnetdb_c,v 1.3 2004/11/26 18:29:15 pvalchev Exp $
--- nsprpub/pr/src/misc/prnetdb.c.orig	Fri May  2 08:17:51 2008
+++ nsprpub/pr/src/misc/prnetdb.c	Mon Aug  3 18:41:06 2009
@@ -2023,7 +2023,12 @@ PR_IMPLEMENT(PRAddrInfo *) PR_GetAddrInf
          */
         hints.ai_socktype = SOCK_STREAM;
 
+	LOCK_DNS();
+
         rv = GETADDRINFO(hostname, NULL, &hints, &res);
+
+	UNLOCK_DNS();
+
         if (rv == 0)
             return (PRAddrInfo *) res;
 
